src/modal/modal-stack.ts

```diff
@@ -25,13 +26,19 @@ export class NgbModalStack {
 
   constructor(
       private _applicationRef: ApplicationRef, private _injector: Injector, @Inject(DOCUMENT) private _document,
-      private _scrollBar: ScrollBar) {}
+      private _scrollBar: ScrollBar, private _rendererFactory: RendererFactory2) {}
 
   open(moduleCFR: ComponentFactoryResolver, contentInjector: Injector, content: any, options): NgbModalRef {
     const containerEl =
         isDefined(options.container) ? this._document.querySelector(options.container) : this._document.body;
+    const renderer = this._rendererFactory.createRenderer(null, null);
```

Comment:
Is there any specific reason you are using `RendererFactory2` here to create a new renderer instance instead of injecting directly `private _renderer: Renderer2` in constructor ?